sweep/fee_bumper.go

```diff
@@ -76,4 +92,436 @@ type BumpResult struct {
 
 	// Err is the error that occurred during the broadcast.
 	Err error
+
+	// requestID is the ID of the request that created this record.
+	requestID uint64
+}
+
+// TxPublisherConfig is the config used to create a new TxPublisher.
+type TxPublisherConfig struct {
+	// Signer is used to create the tx signature.
+	Signer input.Signer
+
+	// Wallet is used primarily to publish the tx.
+	Wallet Wallet
+
+	// Estimator is used to estimate the fee rate for the new tx based on
+	// its deadline conf target.
+	Estimator chainfee.Estimator
+
+	// Notifier is used to monitor the confirmation status of the tx.
+	Notifier chainntnfs.ChainNotifier
+
+	// WeightEstimator is used to calculate the weight of the tx.
+	WeightEstimator input.TxWeightEstimator
+}
+
+// TxPublisher is an implementation of the Bumper interface. It utilizes the
+// `testmempoolaccept` RPC to bump the fee of txns it created based on
+// different fee function selected or configed by the caller. Its purpose is to
+// take a list of inputs specified, and create a tx that spends them to a
+// specified output. It will then monitor the confirmation status of the tx,
+// and if it's not confirmed within a certain time frame, it will attempt to
+// bump the fee of the tx by creating a new tx that spends the same inputs to
+// the same output, but with a higher fee rate. It will continue to do this
+// until the tx is confirmed or the fee rate reaches the maximum fee rate
+// specified by the caller.
+type TxPublisher struct {
+	cfg *TxPublisherConfig
+	wg  sync.WaitGroup
+
+	// currentHeight is the current block height.
+	currentHeight int32
+
+	// records is a map keyed by the txid of the txns being monitored.
+	records lnutils.SyncMap[uint64, *monitorRecord]
+
+	// requestCounter is a monotonically increasing counter used to keep
+	// track of how many requests have been made.
+	requestCounter atomic.Uint64
+
+	// resultChans is a map keyed by the requestCounter, each item is the
+	// chan that the publisher sends the fee bump result to.
+	resultChans lnutils.SyncMap[uint64, chan *BumpResult]
+
+	// quit is used to signal the publisher to stop.
+	quit chan struct{}
+}
+
+// Compile-time constraint to ensure TxPublisher implements Bumper.
+var _ Bumper = (*TxPublisher)(nil)
+
+// NewTxPublisher creates a new TxPublisher.
+func NewTxPublisher(cfg *TxPublisherConfig) *TxPublisher {
```

Comment:
I'd suggest not making this a pointer for the cfg so we can avoid aliasing issues?

Reply:
What is the aliasing issue?

Reply:
If we copy pointers then we expose ourselves to interaction effects. In theory this shouldn't be an issue given that a "Config" is not supposed to be updated, but do we wanna leave that up to chance?

Reply:
cool switched